sdcflows/interfaces/fmap.py

```diff
@@ -23,6 +24,111 @@
+    long_te_phase_metadata = traits.Dict(desc='long TE phase image metadata')
+    phasediff_metadata = traits.Dict(desc='the phasediff metadata')
+
+
+class ProcessPhases(SimpleInterface):
+    """
+    Process phase1, phase2 images so they can be unwrapped.
+
+    Steps are taken from https://fsl.fmrib.ox.ac.uk/fsl/fslwiki/FUGUE/Guide
+    """
+    input_spec = ProcessPhasesInputSpec
+    output_spec = ProcessPhasesOutputSpec
+
+    def _run_interface(self, runtime):
+        images = [self.inputs.phase1_file, self.inputs.phase2_file]
+        metadatas = [self.inputs.phase1_metadata, self.inputs.phase2_metadata]
+        echo_times = [meta.get("EchoTime") for meta in metadatas]
+        if None in echo_times or echo_times[0] == echo_times[1]:
+            raise RuntimeError()
+
+        # Order the images by echo time
+        short_echo_index = echo_times.index(min(echo_times))
+        long_echo_index = echo_times.index(max(echo_times))
+        short_echo_image = images[short_echo_index]
+        long_echo_image = images[long_echo_index]
+
+        # Rescale and save the short TE image
+        se_phase_image = nb.load(short_echo_image)
+        se_phase_data = se_phase_image.get_fdata()
+        se_rescaled_data = rescale_phase_image(se_phase_data)
+        se_rescaled_output = fname_presuffix(short_echo_image, suffix='_shortTE_scaled',
+                                             newpath=runtime.cwd)
+        nb.Nifti1Image(se_rescaled_data, se_phase_image.affine, se_phase_image.header
+                       ).to_filename(se_rescaled_output)
+        self._results['short_te_phase_image'] = se_rescaled_output
+        self._results['short_te_phase_metadata'] = metadatas[short_echo_index]
+
+        # Rescale and save the long TE image
+        le_phase_image = nb.load(long_echo_image)
+        le_phase_data = le_phase_image.get_fdata()
+        le_rescaled_data = rescale_phase_image(le_phase_data)
+        le_rescaled_output = fname_presuffix(long_echo_image, suffix='_longTE_scaled',
+                                             newpath=runtime.cwd)
+        nb.Nifti1Image(le_rescaled_data, le_phase_image.affine, le_phase_image.header
+                       ).to_filename(le_rescaled_output)
+        self._results['long_te_phase_image'] = le_rescaled_output
+        self._results['long_te_phase_metadata'] = metadatas[long_echo_index]
+
+        merged_metadata = deepcopy(metadatas[0])
+        del merged_metadata['EchoTime']
+        merged_metadata['EchoTime1'] = float(echo_times[short_echo_index])
+        merged_metadata['EchoTime2'] = float(echo_times[long_echo_index])
+        self._results['phasediff_metadata'] = merged_metadata
+        return runtime
+
+
+def rescale_phase_image(phase_data):
+    """Ensure that phase images are in a usable range for unwrapping.
+
+    From the FUGUE User guide::
+
+        If you have seperate phase volumes that are in integer format then do:
```

Comment:
please note this is only for integer, could we add a check and raise an error if `phase_data` is not integer? What happens if the image is `int32` and the maximum value is 32k instead of 4096?.

Reply:
The instructions on the FUGUE user guide are vague on this point. They say if the input values don't fit into this range, then do whatever it takes to scale between 0 and 2pi.

Reply:
This scaling happens before phase-unwrapping right?

Reply:
yes, that is correct

Reply:
Not sure if this would ever make a difference in final SDC results, but for integer phase images, the maximum value is usually 4095 because 4096 would wrap back to 0. The proper divisor would be `imax - imin + 1`.

As for non-integer phase images, I have not seen any before (any links to an example?).